custom/fitness-demo/contexts/RecordingProvider.js

```diff
@@ -0,0 +1,337 @@
+    if (!callObject || !isRecordingLocally) return;
+    callObject.startRecording();
+  }, [callObject, isRecordingLocally]);
+
+  useEffect(() => {
+    let timeout;
+    switch (recordingState) {
+      case RECORDING_COUNTDOWN_3:
+        timeout = setTimeout(() => {
+          setRecordingState(RECORDING_COUNTDOWN_2);
+        }, 1000);
+        break;
+      case RECORDING_COUNTDOWN_2:
+        timeout = setTimeout(() => {
+          setRecordingState(RECORDING_COUNTDOWN_1);
+        }, 1000);
+        break;
+      case RECORDING_COUNTDOWN_1:
+        startRecording();
+        break;
+      case RECORDING_ERROR:
+      case RECORDING_SAVED:
+        timeout = setTimeout(() => {
+          setRecordingState(RECORDING_IDLE);
+          setIsRecordingLocally(false);
+        }, 5000);
+        break;
+      default:
+        break;
+    }
+    return () => {
+      clearTimeout(timeout);
+    };
+  }, [recordingState, startRecording]);
+
+  // Show a custom capsule when recording in progress
+  useEffect(() => {
+    if (recordingState !== RECORDING_RECORDING) {
+      setCustomCapsule(null);
+    } else {
+      setCustomCapsule({ variant: 'recording', label: 'Recording' });
+    }
+  }, [recordingState, setCustomCapsule]);
+
+  const startRecordingWithCountdown = useCallback(() => {
+    if (!callObject || !enableRecording) return;
+    setIsRecordingLocally(true);
+    setRecordingState(RECORDING_COUNTDOWN_3);
+    callObject?.sendAppMessage({
+      event: 'recording-starting',
+    });
+  }, [callObject, enableRecording]);
+
+  const stopRecording = useCallback(() => {
+    if (!callObject || !enableRecording || !isRecordingLocally) return;
+    if (recordingState === RECORDING_RECORDING) {
+      switch (enableRecording) {
+        case RECORDING_TYPE_LOCAL:
+        case RECORDING_TYPE_OUTPUT_BYTE_STREAM:
+          setRecordingState(RECORDING_SAVED);
+          setIsRecordingLocally(false);
+          break;
+        case RECORDING_TYPE_CLOUD:
+        case RECORDING_TYPE_CLOUD_BETA:
+        case RECORDING_TYPE_RTP_TRACKS:
+          setRecordingState(RECORDING_UPLOADING);
+          setRecordingState(RECORDING_SAVED);
+          break;
+        default:
+          break;
+      }
+    } else if (recordingState === RECORDING_IDLE) {
+      return;
+    } else {
+      setIsRecordingLocally(false);
+      setRecordingState(RECORDING_IDLE);
+    }
+    setRecordingStartedDate(null);
+    callObject.stopRecording();
+  }, [callObject, enableRecording, isRecordingLocally, recordingState]);
+
+  return (
+    <RecordingContext.Provider
+      value={{
+        isRecordingLocally,
+        recordingStartedDate,
+        recordingState,
+        startRecordingWithCountdown,
+        stopRecording,
+      }}
+    >
+      {children}
+    </RecordingContext.Provider>
+  );
+};
+
+RecordingProvider.propTypes = {
+  children: PropTypes.node,
+};
+
+export const useRecording = () => useContext(RecordingContext);
```

Comment:
yeah, because our RecordingProvider is in the recording example and it's not imported from the `custom/shared`, we can still import it from the example but we can't make the fitness demo as a separate example (without having any links with other demos) or we can do it in another way to move the RecordingProvider to the shared and import it from there.

Reply:
@harshithpabbati you can import from other examples if you add them to the transpile list in`next.config.js`

Reply:
@harshithpabbati let's make a note in the README about interdependency. I think for this demo it makes sense even if it is a bit of anti-pattern

Reply:
@jptaylor yep makes sense, will do the required changes!